src/WP_CLI/CommandWithTerms.php

```diff
@@ -130,7 +133,30 @@ public function remove( $args, $assoc_args ) {
 		if ( $field = Utils\get_flag_value( $assoc_args, 'by' ) ) {
 			$terms = $this->prepare_terms( $field, $terms, $taxonomy );
 		}
-		$result = wp_remove_object_terms( $object_id, $terms, $taxonomy );
+
+		if ( $field = Utils\get_flag_value( $assoc_args, 'all' ) ) {
+
+			// No need to specify terms while removing all terms.
+			if ( $terms ) {
+				WP_CLI::error( 'No need to specify terms while removing all terms.' );
+			}
+
+			// Remove all set categories from post.
+			$result = wp_delete_object_term_relationships( $object_id, $taxonomy );
+
+			// Set post to default category.
+			$cat_id = array( 1 );
```

Comment:
This should only be assigned if the taxonomy is 'category'. Also, the default category is a configurable option we should respect.